Use PSR-17 factory from Guzzle/psr7 2.0

t0mmy742 opened the pull request:
`http-interop/http-factory-guzzle` recommends to use PSR-17 implementation from `guzzlehttp/psr7` if we can use version `^2.0`. Since it matches our dependencies, I replaced old package with the new one.
Users can still use `http-interop/http-factory-guzzle` if they add it manually.

Closes #3105

l0gicgate commented:
This shouldn’t break any downstream projects since they would already have Guzzle PSR-7 which includes the Psr17 factories correct?

t0mmy742 commented:
@l0gicgate Since we dropped support for `guzzlehttp/psr7:^1.0` and updated to `^2.0`, people using this PSR-7 provider already have PSR-17 factory.
If keeping `http-interop/http-factory-guzzle` is something wanted, people already installed it manually since it's not in our requirement packages.
TODO: We should update README file to specify version of supported PSR-7 providers.

ArrayIterator commented:
On version (current / 4.8.1 release)  not yet update.

https://github.com/slimphp/Slim/blob/c8934c35d9d98b1a1df9f99ee69b77a59e0aa820/Slim/Factory/Psr17/GuzzlePsr17Factory.php#L15

Any info about released from current master?

l0gicgate commented:
@ArrayIterator just release 4.9.0 which features the changes.
